m4/mpi.m4

```diff
@@ -184,17 +208,69 @@ AS_IF([test "x$MPI_IMPL" != x],
                     [
                        MPI_IMPL="built-in"
                        AC_MSG_RESULT( [$CXX Compiler Supports MPI] )
-                       AC_DEFINE(HAVE_MPI, 1, [Flag indicating whether or not MPI is available])
                     ],
                     [AC_MSG_RESULT([$CXX Compiler Does NOT Support MPI...]); enablempi=no])
       ])
+     dnl Save variables...
+     AC_SUBST(MPI)
+     AC_SUBST(MPI_IMPL)
+     AC_SUBST(MPI_LIBS)
+     AC_SUBST(MPI_LIBS_PATH)
+     AC_SUBST(MPI_LIBS_PATHS)
+     AC_SUBST(MPI_INCLUDES_PATH)
+     AC_SUBST(MPI_INCLUDES_PATHS)
+
+     VERSION_MPI
+])
+
+
+AC_DEFUN([VERSION_MPI], [
 
-dnl Save variables...
-AC_SUBST(MPI)
-AC_SUBST(MPI_IMPL)
-AC_SUBST(MPI_LIBS)
-AC_SUBST(MPI_LIBS_PATH)
-AC_SUBST(MPI_LIBS_PATHS)
-AC_SUBST(MPI_INCLUDES_PATH)
-AC_SUBST(MPI_INCLUDES_PATHS)
+  AS_IF([test "x$enablempi" != xno],
+  [
+        dnl check that MPI_VERSION and MPI_SUBVERSION are defined.
+        AC_LANG_SAVE
+        AC_LANG_CPLUSPLUS
+        AC_TRY_LINK([@%:@include <mpi.h>],
+                    [#ifndef MPI_VERSION
+                     #error "MPI-Version undefined."
+                    #endif
+                    ],
+                    [ dnl true: Check which version it is:
+                    dnl check that MPI_VERSION is not 1; else disable MPI
+                    AC_TRY_LINK([@%:@include <mpi.h>],
+                                [
+                                #if MPI_VERSION == 1
+                                #error "Bad MPI-version"
+                                #endif
+                                int np; MPI_Comm_size (MPI_COMM_WORLD, &np)
+                                ],
+                                [ dnl it gives true if MPI_VERSION !=1...
+                                 dnl check that MPI_VERSION is not 2; else through a warning
+                                 AC_TRY_LINK([@%:@include <mpi.h>],
+                                             [
+                                             #if MPI_VERSION == 2
```

Comment:
These `#` signs can use the same `@%:@` quadrigraph that is used for includes. Shouldn't make any difference in the generated configure script but it does help syntax highlighting in some editors (emacs).